src/java_bytecode/java_entry_point.cpp

```diff
@@ -245,39 +314,24 @@ void java_record_outputs(
         index_exprt(string_constantt(p_symbol.base_name),
                     gen_zero(index_type())));
       output.op1()=main_arguments[param_number];
-      output.add_source_location()=parameters[param_number].source_location();
+      output.add_source_location()=function.location;
 
       init_code.move_to_operands(output);
     }
   }
 }
 
-/*******************************************************************\
-
-Function: java_entry_point
-
-  Inputs:
-
- Outputs:
-
- Purpose:
-
-\*******************************************************************/
 
-bool java_entry_point(
+main_function_resultt get_main_symbol(
   symbol_tablet &symbol_table,
   const irep_idt &main_class,
   message_handlert &message_handler)
 {
-  // check if the entry point is already there
-  if(symbol_table.symbols.find(goto_functionst::entry_point())!=
-     symbol_table.symbols.end())
-    return false; // silently ignore
+  symbolt symbol;
```

Comment:
was used to either hold the copy of an existing symbol or to serve as empty symbol in case no entry point is found or entry point is ambiguous. Replaced with `symbolt()` in these two cases and locally visible `const symbolt &` when found.